### PR TITLE
Add exp push/pull to getting started guide

### PR DESCRIPTION
**Pull request by dberenbaum**
Adds `dvc exp push` and `dvc exp pull` examples to the getting started guide for experiments. This is a pretty artificial example, and it requires setting up artificial dvc and git remotes. Other suggestions welcome!

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Hi 👋 I see this is a draft but a quick rec is to make the PR base `experiments_reset` for now so the diff is readable (if you want people to check it out while WIP) and change the base back to `master` once #2195 gets merged.

**Review by shcheklein — APPROVED** [review]

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [inline content/docs/start/experiments.md:177]:
```suggestion
After committing the best experiments to our Git branch, we can
[store and share](/doc/start/data-versioning#storing-and-sharing) them
remotely like any other iteration of the pipeline.
```

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [inline content/docs/start/experiments.md:190]: Not sure we need all the Git remote related info? Makes the section quite long too. It's hidden so not a huge deal but it's typically better to avoid explaining other tools, even Git. Even explaining DVC remotes much in this doc is probably unnecessary (already introed in https://dvc.org/doc/start/data-versioning#storing-and-sharing).

Comment (jorgeorpinel) [reply content/docs/start/experiments.md:190]: My rec is to focus on the application of DVC remotes for the `exp` context.

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [inline content/docs/start/experiments.md:218]: > persistent part of the pipeline

Each experiment contains a full pipeline version right? They're not parts of a DVC pipeline I think:

```suggestion
Experiments that have not been made persistent will not
be stored or shared remotely through `dvc push` or `git push`.
```

> Needs reformat

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [reply content/docs/start/experiments.md:190]: That said I may be missing something (didn't read this through in detail) — e.g. that `exp push/pull` use Git remotes. But if it's possible to reduce X remote explanations at least, that would be best I think.

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [inline content/docs/start/experiments.md:235]:
```suggestion
`dvc exp pull` retrieves the experiment from a Git remote.
```

**Review by jorgeorpinel — APPROVED** [review]
Nothing major. The `exp push/pull` part is a bit confusing (Git vs DVC remotes) but I guess that comes from the feature implementation itself.

File: content/docs/start/experiments.md
Comment (dberenbaum) [reply content/docs/start/experiments.md:190]: Yeah, this whole section was a bit of a challenge because:
1) It requires both a Git remote and DVC remote to actually try it out, so I wasn't sure whether to review those concepts so users who come straight to this section can try it, or to ignore them and keep it focused on `exp push/pull` commands only.
2) Without a more complex example of a shared project, which seems out of scope here, there's not much to show the utility of `exp push/pull`.

Comment (dberenbaum) [reply content/docs/start/experiments.md]: I shortened it, but maybe it's still too much? Let me know your thoughts.

Comment (dberenbaum) [reply content/docs/start/experiments.md]: Here's the updated details section to save you some time:

```
### 💡 Expand for more info on remote storage.

The commands in this section require both a default DVC remote and a Git remote.
A DVC remote stores the experiment data, and a Git remote stores the code,
parameters, and other metadata associated with the experiment. DVC supports
various types of remote storage (local file system, SSH, Amazon S3, Google Cloud
Storage, HTTP, HDFS, etc.). The Git remote is often a central Git server
(GitHub, GitLab, BitBucket, etc.).
```

Comment (dberenbaum) [reply content/docs/start/experiments.md]: 👋 @jorgeorpinel, could you review this one edit before I merge?

Comment (dberenbaum) [reply content/docs/start/experiments.md]: @jorgeorpinel Just pinging you again on this one 😄 . Let me know if you have specific concerns or even think this section is unhelpful/doesn't belong in getting started. I'm conflicted about it myself.

Comment (jorgeorpinel) [reply content/docs/start/experiments.md]: Oops @dberenbaum sorry this fell through the cracks, checking now...

Comment (jorgeorpinel) [reply content/docs/start/experiments.md]: Yeah def. very nicely summarized. I'd just add some links perhaps.

File: content/docs/start/experiments.md
Comment (jorgeorpinel) [inline content/docs/start/experiments.md:188]: Idea:
```suggestion
### 💡 Important information on storing experiments remotely.

The commands in this section require both a `dvc remote default` and a
[Git remote](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes).
```

**Comment by jorgeorpinel** [comment]
Still approved 🙂 please don't wait for me to merge, we can always re-edit on another PR. Thanks!

**Pull request merged**
